chef_master/source/deprecations_custom_resource_cleanups.rst

```diff
@@ -0,0 +1,141 @@
+
+In current versions of Chef, ``nil`` was often used to mean that a property had no good default, and needed to be set by the user. However, it is often to useful to set a property to ``nil``, meaning that it's not set and should be ignored. In Chef 13, it is an error to set ``default: nil`` on a property if that property doesn't allow ``nil`` as a valid value.
+
+Remediation
+--------------
+
+If it is valid for the property to be set to nil, then update the property to include that.
+
+.. code-block:: ruby
+
+  property :my_nillable_property, [ String, nil ], default: nil
+
+Otherwise, remove the ``default: nil`` statement from the property.
+
+Invalid Defaults
+==================
+
+Current versions of Chef emit a warning when a property's default value is not valid. This is often because the type of the default value doesn't match the specification of the property. For example:
+
+.. code-block:: ruby
+
+  property :my_property, [ String ], default: []
+
+sets the type of the property to be a String, but then sets the default to be an Array. In Chef 13, this will be an error.
+
+Remediation
+--------------
+
+Ensure that the default value of a property is correct.
+
+Property Getters
+=======================
+
+When writing a resource in Chef 12, calling ``some_property nil`` behaves as a getter, returning the value of ``some_property``. In Chef 13, this will change to set ``some_property`` to ``nil``.
+
+Remediation
+--------------
+
+Simply write ``some_property`` when retrieving the value of ``some_property``.
+
+Specifying both "default" and "name_property" on a resource
+============================================================
+
+Current versions of Chef emit a warning if the property declaration has both ``default`` and ``name_property`` set. In Chef 13, that will become an error. For example:
+
+.. code-block:: ruby
+
+  property :my_property, [ String ], default: [], name_property: true
+
+Remediation
+------------
+
+A property can either have a default, or it can be a "name" property (meaning that it will take the value of the resource's name if not otherwise specified), but not both.
+
+Overriding provides?
+==========================
+
+Some providers override the ``provides?`` method, used to check whether they are a valid provider on the current platform. In Chef 13, this will cause an error if the provider does not also register themselves with the ``provides`` call.
+
+Example
+--------
+
+.. code-block:: ruby
+
+  def provides?
+    true
+  end
+
+Remediation
+------------
+
+.. code-block:: ruby
+
+  provides :my_provider
+
+  def provides?
+    true
+  end
+
+Don't use the updated method
+=============================
+
+The ``updated=(true_or_false)`` method is deprecated and will be removed from Chef 13.
+
+Example
+--------
+
+.. code-block:: ruby
+
+  action :foo do
+    updated = true
+  end
+
+Remediation
+------------
+
+.. code-block:: ruby
+
+  action :foo do
+    updated_by_last_action true
```

Comment:
My feeling is that we should provide the most minimal thing that works. It will be hard to demonstrate rewriting into `converge_by` blocks succinctly and I worry that if the remediation is too hard people simply won't bother.